### DISCUSSION TITLE
Not understanding selections

### DISCUSSION THREAD

**Discussion by ghost** [orbitalquark/textadept#342] [Q&A] [answered]
Hi.

What am I misunderstanding about working with selections?
Thanks for any assistance.

I do;
```
print(buffer.selections) -- prints 2 as I expect
for i = 1, buffer.selections do
  textadept.editing.enclose('<div>', '</div>')
end
```
with buffer containing;
```
a
b
```
a and b are selected separately;

I get;
```
<div>a</div><div></div>
<div>b</div><div></div>
```
instead of the expected:
```
<div>a</div>
<div>b</div>
```

**Comment by orbitalquark** [comment, marked as answer]
`textadept.editing.enclose()` already takes multiple selections into 
account (according to its documentation). You can just call it once. No 
need to iterate over the number of selections.

Note that not all functions operate on all selections.

**Reply by ghost** [reply]
Yes!
Now all of the functionality that I missed in my previous,
difficult to extend, editor is available in Textadept.
Thanks for your assistance, and wonderful product.
